src/supercontrast/provider/handlers/openai_handler.py

```diff
@@ -0,0 +1,260 @@
+import os
+import pydantic
+
+from langchain.output_parsers import PydanticOutputParser
+from langchain.prompts import PromptTemplate
+from langchain_openai import ChatOpenAI
+from openai import OpenAI
+
+from supercontrast.provider.provider_enum import Provider
+from supercontrast.provider.provider_handler import ProviderHandler
+from supercontrast.task import (
+    OCRBoundingBox,
+    OCRRequest,
+    OCRResponse,
+    SentimentAnalysisRequest,
+    SentimentAnalysisResponse,
+    Task,
+    TranslationRequest,
+    TranslationResponse,
+)
+from supercontrast.task.types.transcription_types import (
+    TranscriptionRequest,
+    TranscriptionResponse,
+)
+from supercontrast.utils.audio import load_audio_file
+from supercontrast.utils.image import (
+    get_image_size,
+    load_image_data,
+    process_image_for_llm,
+)
+from supercontrast.utils.text import truncate_text
+
+# Constants
+
+OPENAI_MODEL_NAME = "gpt-4o"
+
+# Task.SENTIMENT_ANALYSIS
+
+
+class SentimentAnalysisOutput(pydantic.BaseModel):
+    score: float = pydantic.Field(
+        description="Sentiment score between -1 (very negative) and 1 (very positive)"
+    )
+
+
+SENTIMENT_ANALYSIS_PROMPT = PromptTemplate(
```

Comment:
I dont think this is how we want to do this.

Llms are bad at modeling continuous variables like a "score"

Reply:
Fair enough, I can recommend that they provide a 0 or 1 score

Reply:
I fixed this for OpenAI and Anthropic, it now gives a 1, 0, or -1 score